src/main/kotlin/com/github/imflog/schema/registry/tasks/download/DownloadTaskAction.kt

```diff
@@ -36,14 +36,37 @@ class DownloadTaskAction(
                 val metadata = getSchemaMetadata(downloadSubject)
                 val outputDir = File(rootDir.toURI()).resolve(downloadSubject.outputPath)
                 outputDir.mkdirs()
-                if (metadataConfiguration.enabled) {
+                val metadataDirectory = if (metadataConfiguration.enabled) {
                     val metadataDirectory = metadataConfiguration.outputPath?.run {
                         File(rootDir.toURI()).resolve(this)
                     } ?: outputDir
                     metadataDirectory.mkdirs()
-                    writeSchemaMetadata(downloadSubject, metadata, metadataDirectory)
+                    metadataDirectory
+                } else {
+                    null
+                }
+
+                if (metadataConfiguration.enabled) {
+                    writeSchemaMetadata(downloadSubject, metadata, metadataDirectory!!)
```

Comment:
I was trying to avoid copying that `val metadataDIrectory = ` block in the place where I have to download references.

With your suggestion to return `outputDir` in else block, the `!!` is not needed anymore.